neo/core/spiketrainlist.py

```diff
@@ -0,0 +1,294 @@
+# -*- coding: utf-8 -*-
+"""
+This module implements :class:`SpikeTrainList`, a pseudo-list
+which supports a multiplexed representation of spike trains
+(all times in a single array, with a second array indicating which
+neuron/channel the spike is from).
+"""
+
+import numpy as np
+import quantities as pq
+from .spiketrain import SpikeTrain, normalise_times_array
+
+
+def is_spiketrain_or_proxy(obj):
+    return isinstance(obj, SpikeTrain) or getattr(obj, "proxy_for", None) == SpikeTrain
+
+
+class SpikeTrainList(object):
+    """
+    This class contains multiple spike trains, and can represent them
+    either as a list of SpikeTrain objects or as a pair of arrays
+    (all spike times in a single array, with a second array indicating which
+    neuron/channel the spike is from).
+
+    A SpikeTrainList object should behave like a list of SpikeTrains
+    for iteration and item access. It is not intended to be used directly
+    by users, but is available as the attribute `spiketrains` of Segments.
+
+    Examples:
+
+        # Create from list of SpikeTrain objects
+
+        >>> stl = SpikeTrainList(items=(
+        ...     SpikeTrain([0.5, 0.6, 23.6, 99.2], units="ms", t_start=0 * pq.ms, t_stop=100.0 * pq.ms),
+        ...     SpikeTrain([ 0.0007, 0.0112], units="s", t_start=0 * pq.ms, t_stop=100.0 * pq.ms),
+        ...     SpikeTrain([ 1100, 88500], units="us", t_start=0 * pq.ms, t_stop=100.0 * pq.ms),
+        ...     SpikeTrain([], units="ms", t_start=0 * pq.ms, t_stop=100.0 * pq.ms),
+        ... ))
+        >>> stl.multiplexed
+        (array([0., 0., 0., 0., 1., 1., 2., 2.]),
+         array([ 0.5,  0.6, 23.6, 99.2,  0.7, 11.2,  1.1, 88.5]) * ms)
+
+        # Create from a pair of arrays
+
+        >>> stl = SpikeTrainList.from_spike_time_array(
+        ...     np.array([0.5, 0.6, 0.7, 1.1, 11.2, 23.6, 88.5, 99.2]),
```

Comment:
label as float is a bit strange. Maube we should force to int, no ?
The boolean mask will be faster and unambiguous.

Reply:
I think this has been solved by removal, @samuelgarcia correct?